src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -302,20 +304,41 @@ Copyright (c) .NET Foundation. All rights reserved.
     ============================================================
                                         _GetAppHostPaths
 
-    Gets the path to the restored apphost, and if the restored apphost
-    was present, computes the path for the destination apphost.
+    Gets the path to apphost (restored via packages or in an apphost pack),
+    and computes the path for the destination apphost.
     ============================================================
      -->
   <Target Name="_GetAppHostPaths"
-          DependsOnTargets="ResolvePackageAssets"
+          DependsOnTargets="_GetAppHostPathsFromAssetFile;_GetAppHostPathFromAppHostPack"
           Condition="'$(UseAppHost)' == 'true' and '$(_IsExecutable)' == 'true'">
+    
+  </Target>
+
+  <Target Name="_GetAppHostPathFromAppHostPack"
+          Condition="'$(UseAppHostFromAssetsFile)' == 'false'">
+
+    <PropertyGroup>
+      <!-- AppHostSourcePath already set by ResolveFrameworkReferences -->
+      <AppHostIntermediatePath>$(IntermediateOutputPath)$(AssemblyName)$([System.IO.Path]::GetExtension($(AppHostSourcePath)))</AppHostIntermediatePath>
+    </PropertyGroup>
+    
+  </Target>
+
+  <Target Name="_GetAppHostPathsFromAssetFile"
+          DependsOnTargets="ResolvePackageAssets"
+          Condition="'$(UseAppHostFromAssetsFile)' == 'true' and '@(_NativeRestoredAppHostNETCore)' != ''">
     <ItemGroup>
       <_NativeAppHostNETCore Include="@(_NativeRestoredAppHostNETCore->'$(IntermediateOutputPath)$(AssemblyName)%(Extension)')" />
```

Comment:
This item now appears to be local to this target.  Perhaps the following for this target instead?

```xml
  <Target Name="_GetAppHostPathsFromAssetFile"
          DependsOnTargets="ResolvePackageAssets"
          Condition="'$(UseAppHostFromAssetsFile)' == 'true' and '@(_NativeRestoredAppHostNETCore)' != ''">

    <NETSdkError Condition="'@(_NativeRestoredAppHostNETCore->Count())' > 1"
                 ResourceName="MultipleFilesResolved"
                 FormatArguments="$(_DotNetAppHostExecutableName)" />

    <PropertyGroup>
      <AppHostSourcePath>@(_NativeRestoredAppHostNETCore)</AppHostSourcePath>
      <AppHostIntermediatePath>@(_NativeRestoredAppHostNETCore->'$(IntermediateOutputPath)$(AssemblyName)%(Extension)')</AppHostIntermediatePath>
    </PropertyGroup>

  </Target>
```

Reply:
Although, the %(FullPath) transform is probably still need...